providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java

```diff
@@ -180,6 +186,17 @@ public FlagdProvider() {
                 fallBackToEnvOrDefault(MAX_EVENT_STREAM_RETRIES_ENV_VAR_NAME, DEFAULT_MAX_EVENT_STREAM_RETRIES));
     }
 
+    public FlagdProvider(OpenTelemetrySdk telemetrySdk) {
```

Comment:
Note - need a refactoring to provide a option builder for construction

Reply:
we could introduce AutoValue here for having also immutability 

https://github.com/google/auto/blob/main/value/userguide/index.md

Reply:
I am opting for a builder pattern-based approach [1] I will do this in a dedicated PR to reduce complexity :) 

[1] - https://github.com/open-feature/java-sdk-contrib/pull/294/files#diff-04582b0607c583d36a31ea656e9229beceab5f2abc3139710831765da374ce89R43-R47